tests/readers/test_googleads_reader.py

```diff
@@ -219,28 +195,44 @@ def test_validate_correct_id(self, valid_input):
         cond = GoogleAdsReader(**self.kwargs).valid_client_customer_id(valid_input)
         assert cond
 
-    @parameterized.expand([
-        ["end_date", {'date_range_type': "CUSTOM_DATE", 'start_date': datetime.date(2019, 1, 1), 'end_date': None}],
-        ["start_date", {'date_range_type': "CUSTOM_DATE", 'start_date': None, 'end_date': datetime.date(2019, 1, 1)}],
-        ["all_dates", {'date_range_type': "CUSTOM_DATE", 'start_date': None, 'end_date': None}],
-    ])
+    @parameterized.expand(
+        [
+            [{"date_range_type": "CUSTOM_DATE", "start_date": datetime.date(2019, 1, 1), "end_date": None}],
+            [{"date_range_type": "CUSTOM_DATE", "start_date": None, "end_date": datetime.date(2019, 1, 1)}],
+            [{"date_range_type": "CUSTOM_DATE", "start_date": None, "end_date": None}],
+        ]
+    )
     @mock.patch.object(GoogleAdsReader, "__init__", mock_googleads_reader)
-    def test_add_invalid_custom_period_to_report_definition(self, name, invalid_parameter):
+    def test_add_invalid_custom_period_to_report_definition(self, invalid_parameter):
         """Test that report definition dateRangeType is replaced by default value
         when no start_date and end_date are provided
         """
-        expected_range_type = DATE_RANGE_TYPE_POSSIBLE_VALUES[0]
+        with self.assertRaises(NoDateDefinitionException):
+            report_definition = self.get_report_definition(invalid_parameter["date_range_type"])
+            temp_kwargs = self.kwargs.copy()
+            temp_kwargs.update(invalid_parameter)
+            GoogleAdsReader(**temp_kwargs).add_period_to_report_definition(report_definition)
 
-        report_definition = self.get_report_definition(invalid_parameter['date_range_type'])
+    @mock.patch.object(GoogleAdsReader, "__init__", mock_googleads_reader)
+    def test_add_inconsistent_period_to_report_definition(self):
+        with self.assertRaises(InconsistentDateDefinitionException):
+            inconsistent_report_definition = {
+                "date_range_type": "YESTERDAY",
+                "start_date": datetime.date(2019, 1, 1),
+                "end_date": datetime.date(2019, 3, 1),
+            }
+            report_definition = self.get_report_definition(inconsistent_report_definition["date_range_type"])
+            temp_kwargs = self.kwargs.copy()
+            temp_kwargs.update(inconsistent_report_definition)
+            GoogleAdsReader(**temp_kwargs).add_period_to_report_definition(report_definition)
```

Comment:
could we parameterise this test? I have the feeling that we call the same method, we just test it raises a different exception against a different set of parameters. It would improve the readability.

Reply:
Yes we can, didn't know we could parametrize with different exceptions, good to know.